# Identifying potential Business opportunities in Bronx Borough, New York City

##  Introduction to the Business idea

The business proposition I would like to take on will be in the realm of providing clients with information on potential business opportunities available in New York in general and Bronx Burough in particular. The information will be based on foursquare data on venues by category that  and location by neighbourhood in food and tourism industry. The potential clients to be targeted will be those that are planning to establish ventures that deal in food and tourism related services in Bronx Burough of New York. To that end, my service will be tailored toward delivering background and general information to clients about potential neighbourhoods which potential clients  will use as initial input into the decision process that will inform their feasibility studies related to establising food and tourism related businesses in Bronx Burough of New York, USA. 

## Description of the dataset to use 
To obtain information that will provide insights to potential clients, the project will use data to be obtained from foursquare application programming interface(API). The data  is based on personal information that visitors of venues make that include what the venue has to offer, category in which the venues falls, personal impressions about the venue, whether the venue was liked or otherwise, tips for others interested in getting the best experience , among others.  My main focus was on  venues that fell into the food  and tourism attraction categories.

## Methodology
The approach used to glean insights from the data included but not limited to: importing packages and dependencies such as numpy, pandas (necessary to manipulate data for easier presentation in arrays and dataframes; json(since foursquare data is in json file format);geocoder application that was used  to generate latitudes and longitudes  (coordinates) based on addresses, request application  which was used to  scrape html data and transform it into text for analysis; matplolib, which was used in plotting the insights gleaned from extracted data; and folium, which was used to draw maps of the locations using geographical coordinates. Insights were clustered into four potential segments using K means machine learning application which is part of  sklearn library. Results were presented which later formed the basis of recommendations for potential clients.

## Presentation of the report
The first step was to import all the libraries deemed import in the downloading, processing, tabulating, mapping and clustering the data; presenting the results, followed by discussion of the results and drawing conclusions and recommendations.



### Importing the packages

In [1]:
import numpy as np # library to handle data in a vectorized manner

import pandas as pd # library for data analsysis
pd.set_option('display.max_columns', None)
pd.set_option('display.max_rows', None)

import json # library to handle JSON files

from geopy.geocoders import Nominatim # convert an address into latitude and longitude values

import requests # library to handle requests
from pandas.io.json import json_normalize # tranform JSON file into a pandas dataframe

### Matplotlib and associated plotting modules
import matplotlib.cm as cm
import matplotlib.colors as colors

### import k-means from clustering stage
from sklearn.cluster import KMeans

!conda install -c conda-forge folium=0.5.0 --yes # uncomment this line if you haven't completed the Foursquare API lab
import folium # map rendering library

print('Libraries imported.')

usage: conda [-h] [-V] command ...
conda: error: unrecognized arguments: # uncomment this line if you haven't completed the Foursquare API lab


Libraries imported.


## Getting the data and taking a sneakpeek at  it

In [2]:
#Get data
York_data = open('newyork_data.json')

In [3]:
with open('newyork_data.json') as nyk_city_data:
    nwyk_data = json.load(nyk_city_data)

In [4]:
nwyk_data

{'type': 'FeatureCollection',
 'totalFeatures': 306,
 'features': [{'type': 'Feature',
   'id': 'nyu_2451_34572.1',
   'geometry': {'type': 'Point',
    'coordinates': [-73.84720052054902, 40.89470517661]},
   'geometry_name': 'geom',
   'properties': {'name': 'Wakefield',
    'stacked': 1,
    'annoline1': 'Wakefield',
    'annoline2': None,
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.84720052054902,
     40.89470517661,
     -73.84720052054902,
     40.89470517661]}},
  {'type': 'Feature',
   'id': 'nyu_2451_34572.2',
   'geometry': {'type': 'Point',
    'coordinates': [-73.82993910812398, 40.87429419303012]},
   'geometry_name': 'geom',
   'properties': {'name': 'Co-op City',
    'stacked': 2,
    'annoline1': 'Co-op',
    'annoline2': 'City',
    'annoline3': None,
    'annoangle': 0.0,
    'borough': 'Bronx',
    'bbox': [-73.82993910812398,
     40.87429419303012,
     -73.82993910812398,
     40.87429419303012]}},
  {'type': 'Feature',
 

In [5]:
neighborhoods_nwyk = nwyk_data['features']
neighborhoods_nwyk

[{'type': 'Feature',
  'id': 'nyu_2451_34572.1',
  'geometry': {'type': 'Point',
   'coordinates': [-73.84720052054902, 40.89470517661]},
  'geometry_name': 'geom',
  'properties': {'name': 'Wakefield',
   'stacked': 1,
   'annoline1': 'Wakefield',
   'annoline2': None,
   'annoline3': None,
   'annoangle': 0.0,
   'borough': 'Bronx',
   'bbox': [-73.84720052054902,
    40.89470517661,
    -73.84720052054902,
    40.89470517661]}},
 {'type': 'Feature',
  'id': 'nyu_2451_34572.2',
  'geometry': {'type': 'Point',
   'coordinates': [-73.82993910812398, 40.87429419303012]},
  'geometry_name': 'geom',
  'properties': {'name': 'Co-op City',
   'stacked': 2,
   'annoline1': 'Co-op',
   'annoline2': 'City',
   'annoline3': None,
   'annoangle': 0.0,
   'borough': 'Bronx',
   'bbox': [-73.82993910812398,
    40.87429419303012,
    -73.82993910812398,
    40.87429419303012]}},
 {'type': 'Feature',
  'id': 'nyu_2451_34572.3',
  'geometry': {'type': 'Point',
   'coordinates': [-73.82780644716412, 

In [6]:
neighborhoods_nwyk[0:20] # Taking a snapshot at New York neighborhoods

[{'type': 'Feature',
  'id': 'nyu_2451_34572.1',
  'geometry': {'type': 'Point',
   'coordinates': [-73.84720052054902, 40.89470517661]},
  'geometry_name': 'geom',
  'properties': {'name': 'Wakefield',
   'stacked': 1,
   'annoline1': 'Wakefield',
   'annoline2': None,
   'annoline3': None,
   'annoangle': 0.0,
   'borough': 'Bronx',
   'bbox': [-73.84720052054902,
    40.89470517661,
    -73.84720052054902,
    40.89470517661]}},
 {'type': 'Feature',
  'id': 'nyu_2451_34572.2',
  'geometry': {'type': 'Point',
   'coordinates': [-73.82993910812398, 40.87429419303012]},
  'geometry_name': 'geom',
  'properties': {'name': 'Co-op City',
   'stacked': 2,
   'annoline1': 'Co-op',
   'annoline2': 'City',
   'annoline3': None,
   'annoangle': 0.0,
   'borough': 'Bronx',
   'bbox': [-73.82993910812398,
    40.87429419303012,
    -73.82993910812398,
    40.87429419303012]}},
 {'type': 'Feature',
  'id': 'nyu_2451_34572.3',
  'geometry': {'type': 'Point',
   'coordinates': [-73.82780644716412, 

## Creating dataframe for New York data

In [7]:
# create an empty dataframw to  use in  populating  neighborhood data

neighborhoods = pd.DataFrame(columns=['Borough', 'Neighborhood', 'Latitude', 'Longitude'])


for data in neighborhoods_nwyk:
    borough = neighborhood_name = data['properties']['borough'] 
    neighborhood_name = data['properties']['name']
    neighborhood_latlon = data['geometry']['coordinates']
    neighborhood_lat = neighborhood_latlon[1]
    neighborhood_lon = neighborhood_latlon[0]
    
    
    neighborhoods = neighborhoods.append({'Borough': borough,
                                          'Neighborhood': neighborhood_name,
                                          'Latitude': neighborhood_lat,
                                          'Longitude': neighborhood_lon}, ignore_index=True)

    
    

In [10]:
neighborhoods.head()

Borough Neighborhood   Latitude  Longitude
0   Bronx    Wakefield  40.894705 -73.847201
1   Bronx   Co-op City  40.874294 -73.829939
2   Bronx  Eastchester  40.887556 -73.827806
3   Bronx    Fieldston  40.895437 -73.905643
4   Bronx    Riverdale  40.890834 -73.912585

In [8]:
print('The dataframe has {} boroughs and {} neighborhoods.'.format(
        len(neighborhoods['Borough'].unique()),
        neighborhoods.shape[0]
     )
   )

The dataframe has 5 boroughs and 306 neighborhoods.


In [9]:
address = 'New York City, NY'
geolocator = Nominatim()
location = geolocator.geocode(address)
latitude = location.latitude
longitude = location.longitude
print('The geograpical coordinate of New York City are {}, {}.'.format(latitude, longitude))

C:\Users\Zubairmap\Anaconda3\Anaconda31\lib\site-packages\ipykernel_launcher.py:2: DeprecationWarning: Using Nominatim with the default "geopy/1.18.1" `user_agent` is strongly discouraged, as it violates Nominatim's ToS https://operations.osmfoundation.org/policies/nominatim/ and may possibly cause 403 and 429 HTTP errors. Please specify a custom `user_agent` with `Nominatim(user_agent="my-application")` or by overriding the default `user_agent`: `geopy.geocoders.options.default_user_agent = "my-application"`. In geopy 2.0 this will become an exception.
  


The geograpical coordinate of New York City are 40.7308619, -73.9871558.


## Generating a map of New york using folium package

In [10]:

# creating map of New York
# create map of New York using latitude and longitude values
newyork_map = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, borough, neighborhood in zip(neighborhoods['Latitude'], neighborhoods['Longitude'], neighborhoods['Borough'], neighborhoods['Neighborhood']):
    label = '{}, {}'.format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(newyork_map)  
    
newyork_map

In [11]:
CLIENT_ID = 'PISLAD2G24MAO0MDXVTSKFIXVIJXWFDLAQYPABPTXPYBITQ5' # your Foursquare ID
CLIENT_SECRET = '01OX3V13KQA3XBWW0OGOMEI0MJ3MKUY22BQEHY1NQE1000I3' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Muyanja Ssenyonga:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Muyanja Ssenyonga:
CLIENT_ID: PISLAD2G24MAO0MDXVTSKFIXVIJXWFDLAQYPABPTXPYBITQ5
CLIENT_SECRET:01OX3V13KQA3XBWW0OGOMEI0MJ3MKUY22BQEHY1NQE1000I3


In [17]:
# htttps://type your answer here
LIMIT = 50

radius = 500
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    neighborhood_latitude, 
    neighborhood_longitude, 
    radius, 
    LIMIT)


## Delving into Bronx Burough for the insights

In [13]:
CLIENT_ID = 'PISLAD2G24MAO0MDXVTSKFIXVIJXWFDLAQYPABPTXPYBITQ5' # your Foursquare ID
CLIENT_SECRET = '01OX3V13KQA3XBWW0OGOMEI0MJ3MKUY22BQEHY1NQE1000I3' # your Foursquare Secret
VERSION = '20180605' # Foursquare API version

print('Muyanja Ssenyonga:')
print('CLIENT_ID: ' + CLIENT_ID)
print('CLIENT_SECRET:' + CLIENT_SECRET)

Muyanja Ssenyonga:
CLIENT_ID: PISLAD2G24MAO0MDXVTSKFIXVIJXWFDLAQYPABPTXPYBITQ5
CLIENT_SECRET:01OX3V13KQA3XBWW0OGOMEI0MJ3MKUY22BQEHY1NQE1000I3


In [14]:
Bronx = neighborhoods[neighborhoods['Borough'] == 'Bronx'].reset_index(drop=True)
Bronx.head(20)

Borough        Neighborhood   Latitude  Longitude
0    Bronx           Wakefield  40.894705 -73.847201
1    Bronx          Co-op City  40.874294 -73.829939
2    Bronx         Eastchester  40.887556 -73.827806
3    Bronx           Fieldston  40.895437 -73.905643
4    Bronx           Riverdale  40.890834 -73.912585
5    Bronx         Kingsbridge  40.881687 -73.902818
6    Bronx            Woodlawn  40.898273 -73.867315
7    Bronx             Norwood  40.877224 -73.879391
8    Bronx      Williamsbridge  40.881039 -73.857446
9    Bronx          Baychester  40.866858 -73.835798
10   Bronx      Pelham Parkway  40.857413 -73.854756
11   Bronx         City Island  40.847247 -73.786488
12   Bronx        Bedford Park  40.870185 -73.885512
13   Bronx  University Heights  40.855727 -73.910416
14   Bronx      Morris Heights  40.847898 -73.919672
15   Bronx             Fordham  40.860997 -73.896427
16   Bronx        East Tremont  40.842696 -73.887356
17   Bronx          West Farms  40.839475 -73.877745
18   Bronx        High  Bridge  40.836623 -73.926102
19   Bronx             Melrose  40.819754 -73.909422

In [15]:
neighborhood_latitude = Bronx.loc[10, 'Latitude'] # neighborhood latitude value
neighborhood_longitude = Bronx.loc[10, 'Longitude'] # neighborhood longitude value

neighborhood_name = Bronx.loc[10, 'Neighborhood'] # neighborhood name

print('Latitude and longitude values of {} are {}, {}.'.format(neighborhood_name, 
                                                               neighborhood_latitude, 
                                                               neighborhood_longitude))

Latitude and longitude values of Pelham Parkway are 40.85741349808865, -73.85475564017999.


In [42]:
LIMIT = 50

radius = 500
url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
    CLIENT_ID, 
    CLIENT_SECRET, 
    VERSION, 
    neighborhood_latitude, 
    neighborhood_longitude, 
    radius, 
    LIMIT)

In [33]:
results = requests.get(url).json()
results

{'meta': {'code': 200, 'requestId': '5c333aa69fb6b714acafabe5'},
 'response': {'suggestedFilters': {'header': 'Tap to show:',
   'filters': [{'name': 'Open now', 'key': 'openNow'},
    {'name': '$-$$$$', 'key': 'price'}]},
  'headerLocation': 'Pelham Parkway',
  'headerFullLocation': 'Pelham Parkway, Bronx',
  'headerLocationGranularity': 'neighborhood',
  'totalResults': 60,
  'suggestedBounds': {'ne': {'lat': 40.86641350708866,
    'lng': -73.84287843417286},
   'sw': {'lat': 40.848413489088635, 'lng': -73.86663284618712}},
  'groups': [{'type': 'Recommended Places',
    'name': 'recommended',
    'items': [{'reasons': {'count': 0,
       'items': [{'summary': 'This spot is popular',
         'type': 'general',
         'reasonName': 'globalInteractionReason'}]},
      'venue': {'id': '51cca5f5498e2220a7f4f0cf',
       'name': 'Peach Wave',
       'location': {'address': '2125 Williamsbridge Rd',
        'crossStreet': 'Pelham Parkway',
        'lat': 40.85605141607515,
        'lng'

## Results

In [43]:
# creating map of Bronx Burough
# create map of New York using latitude and longitude values
Bronx_map = folium.Map(location=[latitude, longitude], zoom_start=10)

# add markers to map
for lat, lng, borough, neighborhood in zip(neighborhoods['Latitude'], neighborhoods['Longitude'], neighborhoods['Borough'], neighborhoods['Neighborhood']):
    label = '{}, {}'.format(neighborhood, borough)
    label = folium.Popup(label, parse_html=True)
    folium.CircleMarker(
        [lat, lng],
        radius=5,
        popup=label,
        color='blue',
        fill=True,
        fill_color='#3186cc',
        fill_opacity=0.7,
        parse_html=False).add_to(Bronx_map)  
    
Bronx_map

 ### A sneakpeek at venue categories in Bronx by name, category, and geographical location. 
 
Foursquare data returned 2375 venues in all that ranged from dessert joints, restaurants, wine shops, fast food stalls, places serving Tacos, among others. 

In [44]:
# a sneakpeek at venue categories in Bronx
# function that extracts the category of the venue
def get_category_type(row):
    try:
        categories_list = row['categories']
    except:
        categories_list = row['venue.categories']
        
    if len(categories_list) == 0:
        return None
    else:
        return categories_list[0]['name']

In [45]:
venues = results['response']['groups'][0]['items']
    
nearby_venues = json_normalize(venues) # flatten JSON

# filter columns
filtered_columns = ['venue.name', 'venue.categories', 'venue.location.lat', 'venue.location.lng']
nearby_venues =nearby_venues.loc[:, filtered_columns]

# filter the category for each row
nearby_venues['venue.categories'] = nearby_venues.apply(get_category_type, axis=1)

# clean columns
nearby_venues.columns = [col.split(".")[-1] for col in nearby_venues.columns]

nearby_venues.head(15)

name          categories        lat        lng
0                   Peach Wave  Frozen Yogurt Shop  40.856051 -73.855706
1                       Enzo's  Italian Restaurant  40.854232 -73.854362
2             New Song's Nails      Cosmetics Shop  40.852911 -73.853428
3            Sorrento Pizzeria         Pizza Place  40.854389 -73.854520
4             Venezia Pizzeria         Pizza Place  40.859812 -73.855224
5   Mr. Q's Chinese Restaurant  Chinese Restaurant  40.855790 -73.855455
6       Kiraku Hibachi & Sushi    Sushi Restaurant  40.853356 -73.853629
7        Ann Clair's Salumeria       Deli / Bodega  40.850046 -73.851796
8            Sal & Doms Bakery        Dessert Shop  40.865377 -73.855236
9                   Patricia's  Italian Restaurant  40.849313 -73.853536
10              Dunkin' Donuts          Donut Shop  40.856380 -73.856090
11       John & Joe's Pizzeria         Pizza Place  40.854520 -73.865824
12                       PS 83              School  40.849135 -73.858823
13            Carvel Ice Cream      Ice Cream Shop  40.855139 -73.855217
14              Emilio's Pizza         Pizza Place  40.848935 -73.854580

In [46]:
print('{} venues were returned by Foursquare.'.format(nearby_venues.shape[0]))

50 venues were returned by Foursquare.


### Nearby venues with the radius of 1000 meters 

In [47]:
def getNearbyVenues(names, latitudes, longitudes, radius=1000):
    
    venues_list=[]
    for name, lat, lng in zip(names, latitudes, longitudes):
        print(name)
            
        # create the API request URL
        url = 'https://api.foursquare.com/v2/venues/explore?&client_id={}&client_secret={}&v={}&ll={},{}&radius={}&limit={}'.format(
            CLIENT_ID, 
            CLIENT_SECRET, 
            VERSION, 
            lat, 
            lng, 
            radius, 
            LIMIT)
            
        # make the GET request
        results = requests.get(url).json()["response"]['groups'][0]['items']
        
        # return only relevant information for each nearby venue
        venues_list.append([(
            name, 
            lat, 
            lng, 
            v['venue']['name'], 
            v['venue']['location']['lat'], 
            v['venue']['location']['lng'],  
            v['venue']['categories'][0]['name']) for v in results])

    nearby_venues = pd.DataFrame([item for venue_list in venues_list for item in venue_list])
    nearby_venues.columns = ['Neighborhood', 
                  'Neighborhood Latitude', 
                  'Neighborhood Longitude', 
                  'Venue', 
                  'Venue Latitude', 
                  'Venue Longitude', 
                  'Venue Category']
    
    return(nearby_venues)

In [48]:
# creating dataframe to contain nearby venues within a radius of 1000 meters 
Bronx_venues = getNearbyVenues(names=Bronx['Neighborhood'],
                                   latitudes=Bronx['Latitude'],
                                   longitudes=Bronx['Longitude']
                                  )

Wakefield
Co-op City
Eastchester
Fieldston
Riverdale
Kingsbridge
Woodlawn
Norwood
Williamsbridge
Baychester
Pelham Parkway
City Island
Bedford Park
University Heights
Morris Heights
Fordham
East Tremont
West Farms
High  Bridge
Melrose
Mott Haven
Port Morris
Longwood
Hunts Point
Morrisania
Soundview
Clason Point
Throgs Neck
Country Club
Parkchester
Westchester Square
Van Nest
Morris Park
Belmont
Spuyten Duyvil
North Riverdale
Pelham Bay
Schuylerville
Edgewater Park
Castle Hill
Olinville
Pelham Gardens
Concourse
Unionport
Edenwald
Claremont Village
Concourse Village
Mount Eden
Mount Hope
Bronxdale
Allerton
Kingsbridge Heights


In [39]:
print(Bronx_venues.shape)
Bronx_venues.head()

(3254, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
0    Wakefield              40.894705              -73.847201   
1    Wakefield              40.894705              -73.847201   
2    Wakefield              40.894705              -73.847201   
3    Wakefield              40.894705              -73.847201   
4    Wakefield              40.894705              -73.847201   

                         Venue  Venue Latitude  Venue Longitude  \
0             Lollipops Gelato       40.894123       -73.845892   
1           Ripe Kitchen & Bar       40.898152       -73.838875   
2  Jackie's West Indian Bakery       40.889283       -73.843310   
3                     Rite Aid       40.896521       -73.844680   
4              Ali's Roti Shop       40.894036       -73.856935   

         Venue Category  
0          Dessert Shop  
1  Caribbean Restaurant  
2  Caribbean Restaurant  
3              Pharmacy  
4  Caribbean Restaurant

There are 2375 neaby venues categorized by neighborhood, venue category and venue coordinates

In [49]:
print(Bronx_venues.shape)
Bronx_venues.tail(10)

(2375, 7)


Neighborhood  Neighborhood Latitude  Neighborhood Longitude  \
2365  Kingsbridge Heights              40.870392              -73.901523   
2366  Kingsbridge Heights              40.870392              -73.901523   
2367  Kingsbridge Heights              40.870392              -73.901523   
2368  Kingsbridge Heights              40.870392              -73.901523   
2369  Kingsbridge Heights              40.870392              -73.901523   
2370  Kingsbridge Heights              40.870392              -73.901523   
2371  Kingsbridge Heights              40.870392              -73.901523   
2372  Kingsbridge Heights              40.870392              -73.901523   
2373  Kingsbridge Heights              40.870392              -73.901523   
2374  Kingsbridge Heights              40.870392              -73.901523   

                                      Venue  Venue Latitude  Venue Longitude  \
2365                Blink Fitness Riverdale       40.877147       -73.905837   
2366  Lehman Center For The Performing Arts       40.872638       -73.893407   
2367                          Panda Express       40.863001       -73.900894   
2368                               Poe Park       40.865502       -73.892300   
2369                     Lot Less Closeouts       40.878270       -73.905265   
2370                 Broadway Pizza & Pasta       40.878822       -73.904494   
2371                     Fordham Restaurant       40.862734       -73.896778   
2372                        Lucille Roberts       40.862049       -73.899981   
2373    Diamante Poblano 1 Bar & Restaurant       40.861872       -73.901596   
2374         Celia's Restaurant Bar & Grill       40.862590       -73.907201   

                 Venue Category  
2365                        Gym  
2366                    Theater  
2367         Chinese Restaurant  
2368                       Park  
2369             Discount Store  
2370                Pizza Place  
2371                      Diner  
2372                        Gym  
2373                 Restaurant  
2374  Latin American Restaurant

In [50]:
Bronx_venues.groupby('Neighborhood').count()

Neighborhood Latitude  Neighborhood Longitude  Venue  \
Neighborhood                                                                
Allerton                                50                      50     50   
Baychester                              50                      50     50   
Bedford Park                            50                      50     50   
Belmont                                 50                      50     50   
Bronxdale                               50                      50     50   
Castle Hill                             41                      41     41   
City Island                             39                      39     39   
Claremont Village                       15                      15     15   
Clason Point                             8                       8      8   
Co-op City                              50                      50     50   
Concourse                               50                      50     50   
Concourse Village                       50                      50     50   
Country Club                            50                      50     50   
East Tremont                            50                      50     50   
Eastchester                             50                      50     50   
Edenwald                                41                      41     41   
Edgewater Park                          41                      41     41   
Fieldston                               47                      47     47   
Fordham                                 50                      50     50   
High  Bridge                            38                      38     38   
Hunts Point                             21                      21     21   
Kingsbridge                             50                      50     50   
Kingsbridge Heights                     31                      31     31   
Longwood                                50                      50     50   
Melrose                                 50                      50     50   
Morris Heights                          50                      50     50   
Morris Park                             50                      50     50   
Morrisania                              50                      50     50   
Mott Haven                              50                      50     50   
Mount Eden                              50                      50     50   
Mount Hope                              50                      50     50   
North Riverdale                         46                      46     46   
Norwood                                 50                      50     50   
Olinville                               50                      50     50   
Parkchester                             50                      50     50   
Pelham Bay                              50                      50     50   
Pelham Gardens                          46                      46     46   
Pelham Parkway                          50                      50     50   
Port Morris                             44                      44     44   
Riverdale                               45                      45     45   
Schuylerville                           50                      50     50   
Soundview                               49                      49     49   
Spuyten Duyvil                          50                      50     50   
Throgs Neck                             31                      31     31   
Unionport                               50                      50     50   
University Heights                      50                      50     50   
Van Nest                                50                      50     50   
Wakefield                               45                      45     45   
West Farms                              47                      47     47   
Westchester Square                      50                      50     50   
Williamsbridge   

With maximum number of  venues set at 50,   8 are in Clason point (the smallest number), 15 in claremont,  21 in Hnts point, 31 in Throgs Neck and Knightsbridge heights, 39 in City Island, 45 each in Wakefield  and Riverdale, 46 each in Pelham Gardens and North Riverdale, 47 in West Frams and Fieldston, 49 in Soundview, and 50 venues each in the remaining  neighborhoods (the largest number).

In [54]:
print('There are {} uniques categories.'.format(len(Bronx_venues['Venue Category'].unique())))

There are 213 uniques categories.


In [55]:
# Analyzing the Uniques venues
Bronx_hotties = pd.get_dummies(Bronx_venues[['Venue Category']], prefix="", prefix_sep="")

# add neighborhood column back to dataframe
Bronx_hotties['Neighborhood'] = Bronx_venues['Neighborhood'] 

# move neighborhood column to the first column
fixed_columns = [Bronx_hotties.columns[-1]] + list(Bronx_hotties.columns[:-1])
Bronx_hotties = Bronx_hotties[fixed_columns]

Bronx_hotties.head()

Zoo Exhibit  Accessories Store  African Restaurant  Airport Tram  \
0            0                  0                   0             0   
1            0                  0                   0             0   
2            0                  0                   0             0   
3            0                  0                   0             0   
4            0                  0                   0             0   

   American Restaurant  Arcade  Arepa Restaurant  Art Gallery  Art Museum  \
0                    0       0                 0            0           0   
1                    0       0                 0            0           0   
2                    0       0                 0            0           0   
3                    0       0                 0            0           0   
4                    0       0                 0            0           0   

   Arts & Crafts Store  Asian Restaurant  Athletics & Sports  Auto Dealership  \
0                    0                 0                   0                0   
1                    0                 0                   0                0   
2                    0                 0                   0                0   
3                    0                 0                   0                0   
4                    0                 0                   0                0   

   Auto Workshop  BBQ Joint  Bagel Shop  Bakery  Bank  Bar  Baseball Field  \
0              0          0           0       0     0    0               0   
1              0          0           0       0     0    0               0   
2              0          0           0       0     0    0               0   
3              0          0           0       0     0    0               0   
4              0          0           0       0     0    0               0   

   Baseball Stadium  Basketball Court  Beach  Beer Bar  Beer Garden  \
0                 0                 0      0         0            0   
1                 0                 0      0         0            0   
2                 0                 0      0         0            0   
3                 0                 0      0         0            0   
4                 0                 0      0         0            0   

   Boat or Ferry  Bookstore  Border Crossing  Botanical Garden  Bowling Alley  \
0              0          0                0                 0              0   
1              0          0                0                 0              0   
2              0          0                0                 0              0   
3              0          0                0                 0              0   
4              0          0                0                 0              0   

   Breakfast Spot  Brewery  Bridge  Buffet  Burger Joint  Burrito Place  \
0               0        0       0       0             0              0   
1               0        0       0       0             0              0   
2               0        0       0       0             0              0   
3               0        0       0       0             0              0   
4               0        0       0       0             0              0   

   Bus Line  Bus Station  Bus Stop  Business Service  Café  Candy Store  \
0         0            0         0                 0     0            0   
1         0            0         0                 0     0            0   
2         0            0         0                 0     0            0   
3         0            0         0                 0     0            0   
4         0            0         0                 0     0            0   

   Caribbean Restaurant  Check Cashing Service  Cheese Shop  \
0                     0                      0            0   
1                     1                      0            0   
2                     1                      0            0   
3                     0                      0            0   
4                     1   

## Discussion
 The list of unique venue categories included : Zoo Exhibit 	Accessories Store 	African Restaurant 	Airport Tram 	American Restaurant 	Arcade 	Arepa Restaurant 	Art Gallery 	Art Museum 	Arts & Crafts Store 	Asian Restaurant 	Athletics & Sports 	Auto Dealership 	Auto Workshop 	BBQ Joint 	Bagel Shop 	Bakery 	Bank 	Bar 	Baseball Field 	Baseball Stadium 	Basketball Court 	Beach 	Beer Bar 	Beer Garden 	Boat or Ferry 	Bookstore 	Border Crossing 	Botanical Garden 	Bowling Alley 	Breakfast Spot 	Brewery 	Bridge 	Buffet 	Burger Joint 	Burrito Place 	Bus Line 	Bus Station 	Bus Stop 	Business Service 	Café 	Candy Store 	Caribbean Restaurant 	Check Cashing Service 	Cheese Shop 	Chinese Restaurant 	Clothing Store 	Cocktail Bar 	Coffee Shop 	Comfort Food Restaurant 	Construction & Landscaping 	Convenience Store 	Cosmetics Shop 	Cuban Restaurant 	Cupcake Shop 	Dance Studio 	Deli / Bodega 	Dentist's Office 	Department Store 	Dessert Shop 	Diner 	Discount Store 	Distillery 	Dive Bar 	Dog Run 	Donut Shop 	Dumpling Restaurant 	Eastern European Restaurant 	Electronics Store 	Empanada Restaurant 	Farmers Market 	Fast Food Restaurant 	Fish & Chips Shop 	Fish Market 	Food 	Food & Drink Shop 	Food Stand 	Food Truck 	French Restaurant 	Fried Chicken Joint 	Frozen Yogurt Shop 	Fruit & Vegetable Store 	Furniture / Home Store 	Garden 	Gas Station 	Gift Shop 	Golf Course 	Gourmet Shop 	Greek Restaurant 	Grocery Store 	Gym 	Gym / Fitness Center 	Harbor / Marina 	Hardware Store 	Health & Beauty Service 	Health Food Store 	Historic Site 	History Museum 	Hobby Shop 	Home Service 	Hookah Bar 	Hot Dog Joint 	Hotel 	Ice Cream Shop 	Indian Restaurant 	Indie Theater 	Intersection 	Italian Restaurant 	Japanese Restaurant 	Jewelry Store 	Juice Bar 	Kids Store 	Kitchen Supply Store 	Latin American Restaurant 	Laundromat 	Lingerie Store 	Liquor Store 	Lounge 	Mac & Cheese Joint 	Market 	Martial Arts Dojo 	Mattress Store 	Medical Center 	Men's Store 	Metro Station 	Mexican Restaurant 	Middle Eastern Restaurant 	Miscellaneous Shop 	Mobile Phone Shop 	Motel 	Movie Theater 	Moving Target 	Music Store 	Music Venue 	Nature Preserve 	Neighborhood 	New American Restaurant 	Nightclub 	Nightlife Spot 	Optical Shop 	Other Great Outdoors 	Other Nightlife 	Paella Restaurant 	Paper / Office Supplies Store 	Park 	Performing Arts Venue 	Peruvian Restaurant 	Pet Store 	Pharmacy 	Photography Studio 	Pizza Place 	Playground 	Plaza 	Pool 	Pool Hall 	Pub 	Public Art 	Rental Car Location 	Restaurant 	River 	Salon / Barbershop 	Sandwich Place 	Scenic Lookout 	School 	Sculpture Garden 	Seafood Restaurant 	Shipping Store 	Shoe Store 	Shopping Mall 	Shopping Plaza 	Smoke Shop 	Snack Place 	Soccer Field 	Soup Place 	Southern / Soul Food Restaurant 	Spa 	Spanish Restaurant 	Sporting Goods Shop 	Sports Bar 	State / Provincial Park 	Steakhouse 	Supermarket 	Supplement Shop 	Sushi Restaurant 	Tapas Restaurant 	Tattoo Parlor 	Tennis Court 	Tennis Stadium 	Thai Restaurant 	Theater 	Theme Park 	Theme Park Ride / Attraction 	Theme Restaurant 	Thrift / Vintage Store 	Toll Plaza 	Track 	Track Stadium 	Trail 	Train Station 	Travel & Transport 	Tunnel 	Vegetarian / Vegan Restaurant 	Video Game Store 	Video Store 	Vietnamese Restaurant 	Volleyball Court 	Warehouse Store 	Waste Facility 	Wine Shop 	Wings Joint 	Women's Store 	Yoga Studio  and 	Zoo. Meanwhile , based on the count of  venues in Bronx by neighborhood, in Clason point has the smallest number (8),  while claremont has 15  there were 21 in Hunts point, 31 in Throgs Neck and Knightsbridge heights, 39 in City Island, 45 each in Wakefield  and Riverdale, 46 each in Pelham Gardens and North Riverdale, 47 in West Frams and Fieldston, 49 in Soundview, and 50 venues each in the remaining  neighborhoods (the largest number).

Meanwhile based on the mean of the spread of 213 unique venues is concentrated in 52 neighbourhoods in Bronx is presented below

In [56]:
Bronx_grouped = Bronx_hotties.groupby('Neighborhood').mean().reset_index()
Bronx_grouped

Neighborhood  Zoo Exhibit  Accessories Store  African Restaurant  \
0              Allerton     0.000000               0.00                0.00   
1            Baychester     0.000000               0.02                0.00   
2          Bedford Park     0.000000               0.00                0.00   
3               Belmont     0.000000               0.00                0.00   
4             Bronxdale     0.000000               0.00                0.00   
5           Castle Hill     0.000000               0.00                0.00   
6           City Island     0.000000               0.00                0.00   
7     Claremont Village     0.000000               0.00                0.00   
8          Clason Point     0.000000               0.00                0.00   
9            Co-op City     0.000000               0.02                0.00   
10            Concourse     0.000000               0.00                0.00   
11    Concourse Village     0.000000               0.00                0.00   
12         Country Club     0.000000               0.00                0.00   
13         East Tremont     0.060000               0.00                0.00   
14          Eastchester     0.000000               0.00                0.00   
15             Edenwald     0.000000               0.00                0.00   
16       Edgewater Park     0.000000               0.00                0.00   
17            Fieldston     0.000000               0.00                0.00   
18              Fordham     0.000000               0.00                0.02   
19         High  Bridge     0.000000               0.00                0.00   
20          Hunts Point     0.000000               0.00                0.00   
21          Kingsbridge     0.000000               0.00                0.00   
22  Kingsbridge Heights     0.000000               0.00                0.00   
23             Longwood     0.000000               0.00                0.00   
24              Melrose     0.000000               0.00                0.00   
25       Morris Heights     0.000000               0.00                0.00   
26          Morris Park     0.000000               0.00                0.00   
27           Morrisania     0.000000               0.00                0.00   
28           Mott Haven     0.000000               0.00                0.00   
29           Mount Eden     0.000000               0.00                0.00   
30           Mount Hope     0.000000               0.00                0.02   
31      North Riverdale     0.000000               0.00                0.00   
32              Norwood     0.000000               0.00                0.00   
33            Olinville     0.000000               0.00                0.00   
34          Parkchester     0.000000               0.00                0.00   
35           Pelham Bay     0.000000               0.00                0.00   
36       Pelham Gardens     0.000000               0.00                0.00   
37       Pelham Parkway     0.000000               0.00                0.00   
38          Port Morris     0.000000               0.00                0.00   
39            Riverdale     0.000000               0.00                0.00   
40        Schuylerville     0.000000               0.00                0.00   
41            Soundview     0.000000               0.00                0.00   
42       Spuyten Duyvil     0.000000               0.00                0.00   
43          Throgs Neck     0.000000               0.00                0.00   
44            Unionport     0.000000               0.00                0.00   
45   University Heights     0.000000               0.00                0.04   
46             Van Nest     0.000000               0.00                0.00   
47            Wakefield     0.000000               0.00                0.00   
48           West Farms     0.042553               0.00                0.00   
49   Westchester Square     0.000000               0.00      

In [57]:
Bronx_grouped.shape

(52, 213)

In [58]:
num_top_venues = 5

for hood in Bronx_grouped['Neighborhood']:
    print("----"+hood+"----")
    temp = Bronx_grouped[Bronx_grouped['Neighborhood'] == hood].T.reset_index()
    temp.columns = ['venue','freq']
    temp = temp.iloc[1:]
    temp['freq'] = temp['freq'].astype(float)
    temp = temp.round({'freq': 2})
    print(temp.sort_values('freq', ascending=False).reset_index(drop=True).head(num_top_venues))
    print('\n')

----Allerton----
                  venue  freq
0  Fast Food Restaurant  0.10
1            Donut Shop  0.08
2           Pizza Place  0.08
3        Sandwich Place  0.06
4           Bus Station  0.06


----Baychester----
              venue  freq
0  Department Store  0.08
1    Clothing Store  0.06
2     Shopping Mall  0.06
3    Discount Store  0.06
4       Pizza Place  0.04


----Bedford Park----
              venue  freq
0             Diner  0.08
1              Park  0.08
2  Botanical Garden  0.08
3       Pizza Place  0.06
4       Supermarket  0.06


----Belmont----
                venue  freq
0  Italian Restaurant  0.28
1       Deli / Bodega  0.10
2              Bakery  0.10
3         Pizza Place  0.10
4        Dessert Shop  0.06


----Bronxdale----
                venue  freq
0         Pizza Place  0.22
1  Italian Restaurant  0.10
2         Coffee Shop  0.06
3  Chinese Restaurant  0.04
4                Bank  0.04


----Castle Hill----
            venue  freq
0        Bus Stop  0.10
1  

                venue  freq
0  Italian Restaurant  0.19
1       Deli / Bodega  0.13
2         Pizza Place  0.06
3                 Bar  0.03
4          Donut Shop  0.03


----Unionport----
                       venue  freq
0                Pizza Place  0.14
1                      Diner  0.06
2             Cosmetics Shop  0.04
3             Ice Cream Shop  0.04
4  Latin American Restaurant  0.04


----University Heights----
                venue  freq
0       Grocery Store  0.16
1         Pizza Place  0.08
2          Shoe Store  0.06
3  Spanish Restaurant  0.06
4         Supermarket  0.06


----Van Nest----
                venue  freq
0         Pizza Place  0.10
1          Donut Shop  0.06
2         Supermarket  0.06
3  Italian Restaurant  0.06
4  Spanish Restaurant  0.04


----Wakefield----
                  venue  freq
0              Pharmacy  0.11
1  Caribbean Restaurant  0.09
2           Supermarket  0.07
3            Donut Shop  0.07
4  Fast Food Restaurant  0.07


----West Farms--

In [59]:
def return_most_common_venues(row, num_top_venues):
    row_categories = row.iloc[1:]
    row_categories_sorted = row_categories.sort_values(ascending=False)
    
    return row_categories_sorted.index.values[0:num_top_venues]

###  Presentation of ten most common venues in Bronx by neighborhood

In [60]:
num_top_venues = 10

indicators = ['st', 'nd', 'rd']

# create columns according to number of top venues
columns = ['Neighborhood']
for ind in np.arange(num_top_venues):
    try:
        columns.append('{}{} Most Common Venue'.format(ind+1, indicators[ind]))
    except:
        columns.append('{}th Most Common Venue'.format(ind+1))

# create a new dataframe
neighborhoods_venues_sorted = pd.DataFrame(columns=columns)
neighborhoods_venues_sorted['Neighborhood'] = Bronx_grouped['Neighborhood']

for ind in np.arange(Bronx_grouped.shape[0]):
    neighborhoods_venues_sorted.iloc[ind, 1:] = return_most_common_venues(Bronx_grouped.iloc[ind, :], num_top_venues)

neighborhoods_venues_sorted

Neighborhood 1st Most Common Venue      2nd Most Common Venue  \
0              Allerton  Fast Food Restaurant                Pizza Place   
1            Baychester      Department Store             Clothing Store   
2          Bedford Park                  Park                      Diner   
3               Belmont    Italian Restaurant                     Bakery   
4             Bronxdale           Pizza Place         Italian Restaurant   
5           Castle Hill              Bus Stop                Pizza Place   
6           City Island       Harbor / Marina         Seafood Restaurant   
7     Claremont Village        Discount Store             Ice Cream Shop   
8          Clason Point                  Park                   Bus Stop   
9            Co-op City            Shoe Store              Shopping Mall   
10            Concourse            Donut Shop                Pizza Place   
11    Concourse Village         Deli / Bodega                Pizza Place   
12         Country Club    Italian Restaurant                Pizza Place   
13         East Tremont            Donut Shop                Pizza Place   
14          Eastchester  Caribbean Restaurant                      Diner   
15             Edenwald           Supermarket       Fast Food Restaurant   
16       Edgewater Park    Italian Restaurant                Pizza Place   
17            Fieldston           Pizza Place         Chinese Restaurant   
18              Fordham           Pizza Place         Italian Restaurant   
19         High  Bridge      Baseball Stadium                     Lounge   
20          Hunts Point                  Park         Seafood Restaurant   
21          Kingsbridge           Pizza Place         Mexican Restaurant   
22  Kingsbridge Heights           Coffee Shop                      Diner   
23             Longwood  Fast Food Restaurant             Sandwich Place   
24              Melrose    Mexican Restaurant                Pizza Place   
25       Morris Heights              Pharmacy                 Food Truck   
26          Morris Park           Pizza Place         Italian Restaurant   
27           Morrisania        Discount Store                   Pharmacy   
28           Mott Haven           Pizza Place                 Donut Shop   
29           Mount Eden           Pizza Place                   Pharmacy   
30           Mount Hope  Fast Food Restaurant              Grocery Store   
31      North Riverdale           Pizza Place                Bus Station   
32              Norwood           Pizza Place                   Pharmacy   
33            Olinville  Fast Food Restaurant                 Donut Shop   
34          Parkchester           Pizza Place  Latin American Restaurant   
35           Pelham Bay        Sandwich Place       Gym / Fitness Center   
36       Pelham Gardens              Pharmacy                 Donut Shop   
37       Pelham Parkway           Pizza Place                 Donut Shop   
38          Port Morris        Baseball Field                Pizza Place   
39            Riverdale           Pizza Place        Japanese Restaurant   
40        Schuylerville           Pizza Place         Italian Restaurant   
41            Soundview  Fast Food Restaurant                Supermarket   
42       Spuyten Duyvil                  Park                Pizza Place   
43          Throgs Neck    Italian Restaurant              Deli / Bodega   
44            Unionport           Pizza Place                      Diner   
45   University Heights         Grocery Store                Pizza Place   
46             Van Nest           Pizza Place                Supermarket   
47            Wakefield              Pharmacy       Caribbean Restaurant   
48           West Farms                   Zoo                 Donut Shop   
49   Westchester Square           Pizza Place                 Donut Shop   
50       Williamsbridge           Pizza Place       Caribbean Restaurant   
51             Woodlawn         Deli / Bodega              

Based on ten most unique venues in the 52 neighborhoods, while most are good for food related business, some are not that much. This includes Coop city, Clarement vilage, Morris heights, Morrisania, high bridge, University heights, and Baychester. Meanwhile as relates to tourism development,  all the 52 neighborhoods have the potential to become very good food , shopping, exercising, walking and jogging, and travelling tourist attractions. 

### Picking through the haystack: Clustering Bronx neighborhoods

In [61]:
# setting  number of clusters
kclusters = 4

Bronx_clustering = Bronx_grouped.drop('Neighborhood', 1)

# run k-means clustering
kmeans = KMeans(n_clusters=kclusters, random_state=0).fit(Bronx_clustering)

# check cluster labels generated for each row in the dataframe
kmeans.labels_[0:10] 

array([0, 3, 1, 3, 3, 1, 1, 2, 2, 0])

In [62]:
Bronx_merged = Bronx

# add clustering labels
Bronx_merged['Cluster Labels'] = kmeans.labels_

# merge toronto_grouped with toronto_data to add latitude/longitude for each neighborhood
Bronx_merged = Bronx_merged.join(neighborhoods_venues_sorted.set_index('Neighborhood'), on='Neighborhood')

Bronx_merged.head(10) # check the last columns!

Borough    Neighborhood   Latitude  Longitude  Cluster Labels  \
0   Bronx       Wakefield  40.894705 -73.847201               0   
1   Bronx      Co-op City  40.874294 -73.829939               3   
2   Bronx     Eastchester  40.887556 -73.827806               1   
3   Bronx       Fieldston  40.895437 -73.905643               3   
4   Bronx       Riverdale  40.890834 -73.912585               3   
5   Bronx     Kingsbridge  40.881687 -73.902818               1   
6   Bronx        Woodlawn  40.898273 -73.867315               1   
7   Bronx         Norwood  40.877224 -73.879391               2   
8   Bronx  Williamsbridge  40.881039 -73.857446               2   
9   Bronx      Baychester  40.866858 -73.835798               0   

  1st Most Common Venue 2nd Most Common Venue 3rd Most Common Venue  \
0              Pharmacy  Caribbean Restaurant  Fast Food Restaurant   
1            Shoe Store         Shopping Mall     Mobile Phone Shop   
2  Caribbean Restaurant                 Diner  Fast Food Restaurant   
3           Pizza Place    Chinese Restaurant         Deli / Bodega   
4           Pizza Place   Japanese Restaurant            Playground   
5           Pizza Place    Mexican Restaurant                Bakery   
6         Deli / Bodega                   Bar                   Pub   
7           Pizza Place              Pharmacy                  Park   
8           Pizza Place  Caribbean Restaurant  Fast Food Restaurant   
9      Department Store        Clothing Store        Discount Store   

  4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
0            Donut Shop           Supermarket   Fried Chicken Joint   
1        Mattress Store                Bakery      Department Store   
2         Shopping Mall            Donut Shop       Supplement Shop   
3                   Bar                 Plaza    Mexican Restaurant   
4                 Plaza                  Park    Mexican Restaurant   
5   Japanese Restaurant           Coffee Shop                 Diner   
6        Discount Store    Italian Restaurant   Rental Car Location   
7            Donut Shop                  Bank        Sandwich Place   
8        Discount Store                Bakery                   Spa   
9         Shopping Mall           Supermarket            Kids Store   

  7th Most Common Venue 8th Most Common Venue 9th Most Common Venue  \
0      Business Service     Mobile Phone Shop    Spanish Restaurant   
1           Pizza Place        Discount Store           Coffee Shop   
2         Women's Store          Burger Joint        Discount Store   
3           Bus Station        Sandwich Place            Donut Shop   
4           Coffee Shop                   Bar                 Diner   
5    Spanish Restaurant          Burger Joint                   Bar   
6          Liquor Store            Donut Shop           Pizza Place   
7  Caribbean Restaurant                 Diner     Mobile Phone Shop   
8        Sandwich Place              Pharmacy           Supermarket   
9                Bakery            Bagel Shop    Italian Restaurant   

  10th Most Common Venue  
0          Deli / Bodega  
1            Supermarket  
2       Asian Restaurant  
3             Steakhouse  
4                   Bank  
5             Donut Shop  
6               Pharmacy  
7     Spanish Restaurant  
8     Seafood Restaurant  
9             Shoe Store

In [63]:
# creating a map to identify clusters
cluster_map = folium.Map(location=[latitude, longitude], zoom_start=11)

# set color scheme for the clusters
x = np.arange(kclusters)
ys = [i+x+(i*x)**2 for i in range(kclusters)]
colors_array = cm.rainbow(np.linspace(0, 1, len(ys)))
rainbow = [colors.rgb2hex(i) for i in colors_array]

# add markers to the map
markers_colors = []
for lat, lon, poi, cluster in zip(Bronx_merged['Latitude'], Bronx_merged['Longitude'], Bronx_merged['Neighborhood'], Bronx_merged['Cluster Labels']):
    label = folium.Popup(str(poi) + ' Cluster ' + str(cluster), parse_html=True)
    folium.CircleMarker(
        [lat, lon],
        radius=5,
        popup=label,
        color=rainbow[cluster-1],
        fill=True,
        fill_color=rainbow[cluster-1],
        fill_opacity=0.7).add_to(cluster_map)
       
cluster_map

## Getting insights from clusters(if any)


Cluster 1

In [64]:
Bronx_merged.loc[Bronx_merged['Cluster Labels'] == 0, Bronx_merged.columns[[1] + list(range(4, Bronx_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue  \
0            Wakefield               0              Pharmacy   
9           Baychester               0      Department Store   
10      Pelham Parkway               0           Pizza Place   
13  University Heights               0         Grocery Store   
14      Morris Heights               0              Pharmacy   
15             Fordham               0           Pizza Place   
23         Hunts Point               0                  Park   
25           Soundview               0  Fast Food Restaurant   
27         Throgs Neck               0    Italian Restaurant   
29         Parkchester               0           Pizza Place   
30  Westchester Square               0           Pizza Place   
32         Morris Park               0           Pizza Place   
33             Belmont               0    Italian Restaurant   
41      Pelham Gardens               0              Pharmacy   
45   Claremont Village               0        Discount Store   
47          Mount Eden               0           Pizza Place   
48          Mount Hope               0  Fast Food Restaurant   
49           Bronxdale               0           Pizza Place   
50            Allerton               0  Fast Food Restaurant   

        2nd Most Common Venue       3rd Most Common Venue  \
0        Caribbean Restaurant        Fast Food Restaurant   
9              Clothing Store              Discount Store   
10                 Donut Shop              Sandwich Place   
13                Pizza Place                  Shoe Store   
14                 Food Truck               Deli / Bodega   
15         Italian Restaurant                       Diner   
23         Seafood Restaurant  Construction & Landscaping   
25                Supermarket              Breakfast Spot   
27              Deli / Bodega                 Pizza Place   
29  Latin American Restaurant          Spanish Restaurant   
30                 Donut Shop                         Bar   
32         Italian Restaurant                 Coffee Shop   
33                     Bakery               Deli / Bodega   
41                 Donut Shop              Sandwich Place   
45             Ice Cream Shop                  Donut Shop   
47                   Pharmacy               Grocery Store   
48              Grocery Store                 Pizza Place   
49         Italian Restaurant                 Coffee Shop   
50                Pizza Place                  Donut Shop   

        4th Most Common Venue 5th Most Common Venue 6th Most Common Venue  \
0                  Donut Shop           Supermarket   Fried Chicken Joint   
9               Shopping Mall           Supermarket            Kids Store   
10         Chinese Restaurant    Italian Restaurant           Supermarket   
13                Supermarket    Spanish Restaurant            Donut Shop   
14                Pizza Place  Fast Food Restaurant  Gym / Fitness Center   
15  Latin American Restaurant         Deli / Bodega    Chinese Restaurant   
23              Grocery Store           Coffee Shop              Bus Stop   
25             Clothing Store    Chinese Restaurant                  Park   
27           Asian Restaurant            Donut Shop             Bookstore   
29                      Diner        Ice Cream Shop                   Gym   
30       Fast Food Restaurant              Pharmacy        Sandwich Place   
32                   Pharmacy           Supermarket                   Bar   
33                Pizza Place          Dessert Shop                   Bar   
41                Pizza Place    Chinese Restaurant   Rental Car Location   
45                Pizza Place  Fast Food Restaurant                   Gym   
47                 Donut Shop            Food Truck         Deli / Bodega   
48                Supermarket              Pharmacy            Donut Shop   
49         Chinese Restaurant                  Park         Deli / Bodega   
50             Sandwich Place           Bus S

Cluser 2

In [65]:
Bronx_merged.loc[Bronx_merged['Cluster Labels'] == 1, Bronx_merged.columns[[1] + list(range(4, Bronx_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
2   Eastchester               1  Caribbean Restaurant                 Diner   
5   Kingsbridge               1           Pizza Place    Mexican Restaurant   
6      Woodlawn               1         Deli / Bodega                   Bar   
19      Melrose               1    Mexican Restaurant           Pizza Place   
20   Mott Haven               1           Pizza Place            Donut Shop   
39  Castle Hill               1              Bus Stop           Pizza Place   
42    Concourse               1            Donut Shop           Pizza Place   

   3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
2   Fast Food Restaurant         Shopping Mall            Donut Shop   
5                 Bakery   Japanese Restaurant           Coffee Shop   
6                    Pub        Discount Store    Italian Restaurant   
19                   Gym    Italian Restaurant                 Diner   
20    Mexican Restaurant        Discount Store           Supermarket   
39           Bus Station        Baseball Field         Deli / Bodega   
42                   Gym                  Park        Sandwich Place   

   6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
2        Supplement Shop         Women's Store          Burger Joint   
5                  Diner    Spanish Restaurant          Burger Joint   
6    Rental Car Location          Liquor Store            Donut Shop   
19        Discount Store            Donut Shop     Mobile Phone Shop   
20         Grocery Store                   Gym    Spanish Restaurant   
39             Wine Shop    Spanish Restaurant   Fried Chicken Joint   
42  Fast Food Restaurant              Pharmacy                 Diner   

   9th Most Common Venue 10th Most Common Venue  
2         Discount Store       Asian Restaurant  
5                    Bar             Donut Shop  
6            Pizza Place               Pharmacy  
19   Fried Chicken Joint                 Bakery  
20               Brewery             Steakhouse  
39          Liquor Store                 Market  
42        Discount Store          Grocery Store

Clster 3

In [66]:
Bronx_merged.loc[Bronx_merged['Cluster Labels'] == 2, Bronx_merged.columns[[1] + list(range(4, Bronx_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue 2nd Most Common Venue  \
7         Norwood               2           Pizza Place              Pharmacy   
8  Williamsbridge               2           Pizza Place  Caribbean Restaurant   

  3rd Most Common Venue 4th Most Common Venue 5th Most Common Venue  \
7                  Park            Donut Shop                  Bank   
8  Fast Food Restaurant        Discount Store                Bakery   

  6th Most Common Venue 7th Most Common Venue 8th Most Common Venue  \
7        Sandwich Place  Caribbean Restaurant                 Diner   
8                   Spa        Sandwich Place              Pharmacy   

  9th Most Common Venue 10th Most Common Venue  
7     Mobile Phone Shop     Spanish Restaurant  
8           Supermarket     Seafood Restaurant

Clster 4

In [67]:
Bronx_merged.loc[Bronx_merged['Cluster Labels'] == 3, Bronx_merged.columns[[1] + list(range(4, Bronx_merged.shape[1]))]]

Neighborhood  Cluster Labels 1st Most Common Venue  \
1            Co-op City               3            Shoe Store   
3             Fieldston               3           Pizza Place   
4             Riverdale               3           Pizza Place   
11          City Island               3       Harbor / Marina   
12         Bedford Park               3                  Park   
16         East Tremont               3            Donut Shop   
17           West Farms               3                   Zoo   
18         High  Bridge               3      Baseball Stadium   
21          Port Morris               3        Baseball Field   
22             Longwood               3  Fast Food Restaurant   
24           Morrisania               3        Discount Store   
26         Clason Point               3                  Park   
28         Country Club               3    Italian Restaurant   
31             Van Nest               3           Pizza Place   
34       Spuyten Duyvil               3                  Park   
35      North Riverdale               3           Pizza Place   
36           Pelham Bay               3        Sandwich Place   
37        Schuylerville               3           Pizza Place   
38       Edgewater Park               3    Italian Restaurant   
40            Olinville               3  Fast Food Restaurant   
43            Unionport               3           Pizza Place   
44             Edenwald               3           Supermarket   
46    Concourse Village               3         Deli / Bodega   
51  Kingsbridge Heights               3           Coffee Shop   

   2nd Most Common Venue 3rd Most Common Venue 4th Most Common Venue  \
1          Shopping Mall     Mobile Phone Shop        Mattress Store   
3     Chinese Restaurant         Deli / Bodega                   Bar   
4    Japanese Restaurant            Playground                 Plaza   
11    Seafood Restaurant         Boat or Ferry    Italian Restaurant   
12                 Diner      Botanical Garden           Supermarket   
16           Pizza Place           Zoo Exhibit                  Bank   
17            Donut Shop   Fried Chicken Joint                  Park   
18                Lounge                 Plaza        Sandwich Place   
21           Pizza Place            Donut Shop            Restaurant   
22        Sandwich Place        Discount Store            Donut Shop   
24              Pharmacy            Donut Shop  Fast Food Restaurant   
26              Bus Stop                 Trail  Gym / Fitness Center   
28           Pizza Place        Sandwich Place                Bakery   
31           Supermarket    Italian Restaurant            Donut Shop   
34           Pizza Place              Pharmacy                  Bank   
35           Bus Station                  Pool           Coffee Shop   
36  Gym / Fitness Center     Convenience Store    Italian Restaurant   
37    Italian Restaurant              Pharmacy                   Bar   
38           Pizza Place         Deli / Bodega                   Bar   
40            Donut Shop        Sandwich Place           Pizza Place   
43                 Diner            Donut Shop        Sandwich Place   
44  Fast Food Restaurant  Caribbean Restaurant   Fried Chicken Joint   
46           Pizza Place    Mexican Restaurant                   Gym   
51                 Diner           Supermarket                   Gym   

        5th Most Common Venue   6th Most Common Venue 7th Most Common Venue  \
1                      Bakery        Department Store           Pizza Place   
3                       Plaza      Mexican Restaurant           Bus Station   
4                        Park      Mexican Restaurant           Coffee Shop   
11                        Bar  Thrift / Vintage Store   American Restaurant   
12                     Garden             Pizza Place    Mexican Restaurant   
16                       Park          Sandwich Place            Shoe Store   
17                Pizza Place  

## Conclusion
Bronx Burough is bustling location for food and tourism industry.  Its 213 unique venues are spread in 52 neighbourhoods with most of which have the potential to become vibrant locations for food and tourism related businesses. Based on clusters generated, it is apparent that cluster one can be described as 'tribal' or exotic cousine food cluster; cluster two, the shopping and food cluster, cluster three has  the credentials of being a hefty food and entertainment segment, while cluster four deserves to become Asian and Mexican food and tourism attraction segment. There is need to identify key factors that explain the relartively small number of unique venues in Clason point (8), Claremont(15) and Huntspoint(21) compared to the average of 50 venues in most neighborhoods. This could be potential entry point for new food and tourism attraction related business if the problem is lack of facilities that support the two related and interdepedent sectors. Nonethless, it could also shed light on obstacles that have prevented the establishment of vibrant and popular food and tourism attraction venues in such neighborhoods which could save potential investors from investing in neighborhood that have the potential to generate low return in investment.

 ### Work is as good as done: What an illuminating assignment!